packages/psss-server/src/routes/reports/ConfirmedCountryWeeklyReportRoute.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { Route } from '../Route';
+
+export class ConfirmedCountryWeeklyReportRoute extends Route {
+  async buildResponse() {
+    const { startWeek, endWeek } = this.req.query;
+    const { organisationUnitCode: countryCode } = this.req.params;
+    const reportData = await this.reportConnection?.fetchReport(
```

Comment:
Sorry for more back and forth but I just thought of error handling here. Do we want to handle anything specifically, or is the generic error handling wrapped around all endpoints enough?

Reply:
For example, if the report-server sends back a specific error code, perhaps we want to pass that same error code on to the front end? Not sure though...let me know your thoughts and I can return with a fresh brain tomorrow!

Reply:
That sounds very useful! I guess we can do this at the ApiConnection level so that it's available to all the Api connections. I had a look at the `verifyResponseStatus` util that you used in ApiConnection and maybe we can change it to something like the below?
```
const throwCustomError = (status, errorMessage) => {
  const statusCode = status || 500;
  throw new CustomError({
    responseStatus: statusCode,
    responseText: errorMessage,
  });
};

export const verifyResponseStatus = async response => {
  if (!response.ok) {
    let responseJson;
    try {
      responseJson = await response.json();
    } catch (error) {
      throw new Error(`Network error ${response.status}`);
    }
    if (
      response.status &&
      (response.status < 200 || response.status >= 300) &&
      !responseJson.error
    ) {
      throwCustomError(response.status, responseJson.message);
    }
    if (responseJson.error) {
      throwCustomError(response.status, responseJson.error);
    }
  }
};
```

Reply:
Sounds great!

Reply:
As discussed, will resolve this in another PR later!